### ISSUE TITLE
RAP Board

### ISSUE THREAD

**Issue by ryan-d-williams** [ryan-d-williams/Gonkulator_Issues#45]
V1 of the RAP board is now complete

The RAP board is an auto-updating interface to track squadron RAP progress as you schedule. It displays and tracks:
- Completed flights / sims
- What is left to be complete
- Total event progress
- Overflights / over-sims
- Basic metrics

Demo:
![sample44](https://user-images.githubusercontent.com/65094007/148710839-24e59bc9-1e48-4bc1-affb-ed8de873c92e.gif)

How to use it:
The RAP data is based on user-defined RAP settings available via the pilot management page. To use the RAP board, first go through and add RAP data for each of the pilots (should be minimal work):
![image](https://user-images.githubusercontent.com/65094007/148710863-14fcc076-b4ad-490d-9666-4b74ddf862a1.png)

Then while scheduling, click on the RAP board link in the menu. Note that the RAP board initially pulls data from the cloud. So if you have local changes that are not saved, they will not initially be displayed in the RAP board. Once you open the RAP board and see a green "connected" in the top right hand corner, all changes made in the puck throw view will be repeated in the RAP board. You will see a textual reminder of this when you try to open the RAP board
TL;DR: save to cloud before you open the RAP board. 

A green line at the bottom of an event means it already happened (has a date in the past). This is useful to quickly identify what has already happened, and what is scheduled. A white line at the top of an event means it is an overfly or oversim (more than required for RAP.
The rest of the interface should be self explanatory, but please reach out if you have any questions.

Future:
A lot of effort will continue to go into the RAP board. Advanced metrics, such as:
- 3 month lookback
- Pilot availability for the rest of the month based on commitments
- Defining turn patters
Etc, will be added as time goes on. This will also be the beginning of soon-to-be auto-scheduling based on rap metrics.
